doc/design/diagrams/request_response_server_port.puml

```diff
@@ -0,0 +1,37 @@
+@startuml
+skinparam backgroundColor #EEEBDC
+
+class ServerPortData {
+  chunkSenderData: ChunkSenderData
+  chunkReceiverData: ChunkReceiverData
+  offeringRequested: atomic<bool>
+  offered: atomic<bool>
+}
+
+class ServerPortUser {
+  offer(): void
+  stopOffer(): void
+  isOffered(): bool
+  hasClients(): bool
+  getRequest(): expected<const RequestHeader*, ChunkReceiveResult>
+  releaseRequest(response Header: const RequestHeader*): void
+  hasNewRequests(): void
+  hasLostRequestsSinceLastCall(): bool
+  allocateResponse(userPayloadSize: uint32_t, userPayloadAlignment: uint32_t): expected<ResponseHeader*, AllocationError>
+  sendResponse(requestHeader: ResponseHeader): void
+  freeResponset(requestHeader: ResponseHeader): void
```

Comment:
typo: `freeResponse`

Reply:
and I also have a C&P error at `requestHeader: ResponseHeader`